rules/plists.bzl

```diff
@@ -28,38 +28,33 @@ def process_infoplists(name, infoplists, infoplists_by_build_setting, xcconfig,
     # Final substituted infoplists_by_build_setting dict
     substituted_infoplists_by_build_setting = {}
 
-    # Default infoplists are any defined in `infoplists` OR in `infoplists_by_build_setting` with the key `//conditions:default`
-    default_infoplists = infoplists_by_build_setting.pop("//conditions:default", infoplists)
-
-    # Default xcconfig is the one defined in `xcconfig` OR in `xcconfig_by_build_setting` with the key `//conditions:default`
-    default_xcconfig = xcconfig_by_build_setting.pop("//conditions:default", xcconfig)
-
-    # Substitute the default infoplists
-    substituted_infoplists_by_build_setting["//conditions:default"] = [
-        substituted_plist(
-            name = "_%s.%s.info" % (name, idx),
-            plist = plist,
-            xcconfig = default_xcconfig,
-        )
-        for idx, plist in enumerate(write_info_plists_if_needed(name = name, plists = default_infoplists))
-    ]
+    # Default infoplists are any defined in `infoplists`
+    default_infoplists = infoplists if infoplists else []
+
+    # Default xcconfig is the one defined in `xcconfig`
+    default_xcconfig = xcconfig if xcconfig else {}
 
     # Collect a set of config settings to iterate over
-    config_setting_names = sets.make(infoplists_by_build_setting.keys() + xcconfig_by_build_setting.keys())
+    merged_xcconfigs = {}
+    for config_setting_name in xcconfig_by_build_setting.keys():
+        xccconfig_for_build_setting = xcconfig_by_build_setting.get(config_setting_name, {})
+        merged_xcconfigs[config_setting_name]  = merge_xcconfigs(default_xcconfig, xccconfig_for_build_setting)
+
+    # If someone hasn't specified "//conditions:default" - inject it there as upstream callers will `select` on this
+    if len(infoplists_by_build_setting.keys()) == 0:
+        infoplists_by_build_setting["//conditions:default"] = default_infoplists
```

Comment:
I guess the confusing part is this differs from how `xcconfig` and `xcconfig_by_build_setting` work. 

`xcconfig` gets merged into all the different ones but this seems to ignore `infoplists` entirely if `infoplists_by_build_setting` is set

Reply:
If we dont want users to use them both at the same time (cash doesn't use `infoplists_by_build_setting` at all) should we just fail instead?